src/pool/pool.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { InjectRepository } from '@nestjs/typeorm';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
+
+import gql from 'graphql-tag';
+import { Repository } from 'typeorm';
+import fetch from 'cross-fetch';
+
+import { Pool } from './pool.entity';
+import Big from 'big.js';
+
+export const createClient = (uri: string) =>
+  new ApolloClient({
+    link: new HttpLink({ uri, fetch: fetch }),
+    cache: new InMemoryCache(),
+  });
+
+export const HOUR_IN_SECONDS = 60 * 60;
+export const HOURS_IN_DAY = 24;
+
+export const createQuery = (
+  blockTimestamp_lte: string | number,
+  blockTimestamp_gte: string | number,
+) => gql`
+  query Swaps {
+    swaps(
+      first: 1000,
+      where: {
+        blockTimestamp_gte: ${blockTimestamp_gte.toString()}
+        blockTimestamp_lte: ${blockTimestamp_lte.toString()}
+      }
+      orderBy: blockTimestamp
+      orderDirection: asc
+    ) {
+      id
+      tokenIn
+      tokenInAmount
+      poolId
+      tokenOut
+      tokenOutAmount
+      receiptId
+      blockTimestamp
+    }
+  }
+`;
+
+@Injectable()
+export class PoolService {
+  private readonly logger = new Logger(PoolService.name);
+
+  constructor(
+    @InjectRepository(Pool)
+    private readonly poolRepo: Repository<Pool>,
+  ) {}
+
+  @Cron('* * * * *')
+  async handleCron() {
+    this.logger.verbose('handleCron for pool service');
+
+    try {
+      const swaps = await this.requestDataForDay();
+      swaps.map((array) =>
+        this.logger.log(`Results length per hour ${array.length}`),
+      );
+      const flatSwaps = swaps.flat();
+      const newPools: { [key: string]: Pool } = flatSwaps.reduce(
+        (
+          acc: { [key: string]: Pool },
+          swap: {
+            id: string;
+            output: string;
+            poolId: number;
+            blockTimestamp: number;
+            tokenInAmount: number;
+            tokenIn: string;
+            tokenOutAmount: number;
+            tokenOut: string;
+            receiptId: string;
+            predecessorId: string;
+          },
+        ) => {
+          const [, poolId] = swap.id?.split(' ');
```

Comment:
Please add exception handling in this block to avoid a situation when one bad record can crash processing for all pools.